libraries/shared/logs/extractor.go

```diff
@@ -0,0 +1,156 @@
+// VulcanizeDB
+// Copyright © 2019 Vulcanize
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package logs
+
+import (
+	"errors"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/sirupsen/logrus"
+	"github.com/vulcanize/vulcanizedb/libraries/shared/constants"
+	"github.com/vulcanize/vulcanizedb/libraries/shared/fetcher"
+	"github.com/vulcanize/vulcanizedb/libraries/shared/transactions"
+	"github.com/vulcanize/vulcanizedb/libraries/shared/transformer"
+	"github.com/vulcanize/vulcanizedb/pkg/core"
+	"github.com/vulcanize/vulcanizedb/pkg/datastore"
+)
+
+var ErrNoWatchedAddresses = errors.New("no watched addresses configured in the log extractor")
+
+const (
+	missingHeadersFound   = true
+	noMissingHeadersFound = false
+)
+
+type ILogExtractor interface {
+	AddTransformerConfig(config transformer.EventTransformerConfig) error
+	ExtractLogs(recheckHeaders constants.TransformerExecution) (error, bool)
```

Comment:
Yeah, I suppose I was thinking this is sort of a weird case because the bool functions similarly to `ok` in `result, ok := mapping[key]`. Happy to switch up the order if you think that'd make it more clear though

Reply:
Cool, can do